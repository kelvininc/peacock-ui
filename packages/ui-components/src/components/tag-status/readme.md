# kv-tag-status



<!-- Auto Generated Below -->


## Usage

### Angular

```html
<!-- Default -->
<kv-tag-status />

<!-- With Type -->
<kv-tag-status [type]="ETagStatusType.Running" />
```


### React

```tsx
import React from 'react';

import { KvTagStatus, ETagStatusType } from '@kelvininc/react-ui-components';

export const TagStatusExample: React.FC = () => (
	<>
		{/*-- Default --*/}
		<KvTagStatus />

		{/*-- With Type --*/}
		<KvTagStatus type={ETagStatusType.Running} />
	</>
);
```



## Properties

| Property | Attribute | Description                             | Type                                                                                                                                                                                 | Default                  |
| -------- | --------- | --------------------------------------- | ------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------ | ------------------------ |
| `type`   | `type`    | (optional) Defines the status tag type. | `ETagStatusType.Failed \| ETagStatusType.Offline \| ETagStatusType.Online \| ETagStatusType.Partially \| ETagStatusType.Pending \| ETagStatusType.Running \| ETagStatusType.Unknown` | `ETagStatusType.Unknown` |


## Dependencies

### Depends on

- [kv-icon](../icon)

### Graph
```mermaid
graph TD;
  kv-tag-status --> kv-icon
  style kv-tag-status fill:#f9f,stroke:#333,stroke-width:4px
```

----------------------------------------------


