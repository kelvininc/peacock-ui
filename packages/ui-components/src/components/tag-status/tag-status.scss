@import '../../assets/styles/globals';

.status {
	&-container {
		display: flex;
		align-items: center;

		kv-icon {
			--icon-width: 16px;
			--icon-height: 16px;
		}
	}

	&-success {
		kv-icon {
			--icon-color: #{kv-color('success')};
		}
	}

	&-error {
		kv-icon {
			--icon-color: #{kv-color('error')};
		}
	}

	&-none {
		kv-icon {
			--icon-color: #{kv-color('neutral-4')};
		}
	}

	&-warning {
		kv-icon {
			--icon-color: #{kv-color('warning')};
		}
	}
}

.status-label {
	@include kv-font-label-small-regular();
	color: kv-color('neutral-2');
	margin-left: $spacing-2x;
}
