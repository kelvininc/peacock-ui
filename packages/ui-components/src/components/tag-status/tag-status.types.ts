import { EIconName, EOtherIconName } from '../icon/icon.types';

export interface ITagStatus {
	/** (required) Defines the color of the icon.*/
	state: ETagState;
	/** (required) Defines the icon to be displayed.*/
	icon: EIconName | EOtherIconName;
	/** (optional) Defines the content of the label.*/
	label?: string;
}

export enum ETagState {
	Unknown = 'unknown',
	Success = 'success',
	Warning = 'warning',
	Error = 'error',
	Info = 'info'
}
