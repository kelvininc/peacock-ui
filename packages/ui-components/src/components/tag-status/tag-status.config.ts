import { EIconName } from '../icon/icon.types';
import { ETagState, ETagStatusType, ITagStatusConfig } from './tag-status.types';

export const STATUS_CONFIG: { [key in ETagStatusType]: ITagStatusConfig } = {
	[ETagStatusType.Running]: {
		label: 'Running',
		icon: EIconName.Success,
		state: ETagState.Success
	},
	[ETagStatusType.Online]: {
		label: 'Online',
		icon: EIconName.Success,
		state: ETagState.Success
	},
	[ETagStatusType.Failed]: {
		label: 'Failed',
		icon: EIconName.Error,
		state: ETagState.Error
	},
	[ETagStatusType.Offline]: {
		label: 'Offline',
		icon: EIconName.Error,
		state: ETagState.Error
	},
	[ETagStatusType.Unknown]: {
		label: 'Unknown',
		icon: EIconName.Error,
		state: ETagState.None
	},
	[ETagStatusType.Pending]: {
		label: 'Pending Deploy',
		icon: EIconName.Building,
		state: ETagState.Warning
	},
	[ETagStatusType.Partially]: {
		label: 'Partially Online',
		icon: EIconName.Building,
		state: ETagState.Warning
	}
};
