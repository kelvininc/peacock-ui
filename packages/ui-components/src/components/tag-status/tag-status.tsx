import { Component, Host, h, Prop } from '@stencil/core';
import { STATUS_CONFIG } from './tag-status.config';
import { ETagStatusType, ITagStatus } from './tag-status.types';

@Component({
	tag: 'kv-tag-status',
	styleUrl: 'tag-status.scss',
	shadow: true
})
export class KvTagStatus implements ITagStatus {
	@Prop({ reflect: true }) type: ETagStatusType = ETagStatusType.Unknown;

	render() {
		const { state, icon, label } = STATUS_CONFIG[this.type];

		return (
			<Host>
				<div class={`status-container status-${state}`}>
					<kv-icon name={icon} />
					<span class="status-label">{label}</span>
				</div>
			</Host>
		);
	}
}
