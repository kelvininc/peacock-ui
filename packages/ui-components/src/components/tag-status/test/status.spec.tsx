import { SpecPage } from '@stencil/core/internal';
import { KvTagStatus } from '../tag-status';
import { newSpecPage } from '@stencil/core/testing';

describe('Status Tag (unit tests)', () => {
	let page: SpecPage;

	describe('when uses default props', () => {
		beforeEach(async () => {
			page = await newSpecPage({
				components: [KvTagStatus],
				html: `<kv-tag-status></kv-tag-status>`
			});
		});

		it('should match the snapshot', () => {
			expect(page.root).toMatchSnapshot();
		});
	});

	describe('when has a type', () => {
		beforeEach(async () => {
			page = await newSpecPage({
				components: [KvTagStatus],
				html: `<kv-tag-status type="pending"></kv-tag-status>`
			});
		});

		it('should match the snapshot', () => {
			expect(page.root).toMatchSnapshot();
		});
	});
});
