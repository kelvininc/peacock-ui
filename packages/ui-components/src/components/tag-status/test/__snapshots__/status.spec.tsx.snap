// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Status Tag (unit tests) when has a type should match the snapshot 1`] = `
<kv-tag-status type="pending">
  <mock:shadow-root>
    <div class="status-container status-warning">
      <kv-icon name="kv-building"></kv-icon>
      <span class="status-label">
        Pending Deploy
      </span>
    </div>
  </mock:shadow-root>
</kv-tag-status>
`;

exports[`Status Tag (unit tests) when uses default props should match the snapshot 1`] = `
<kv-tag-status type="unknown">
  <mock:shadow-root>
    <div class="status-container status-none">
      <kv-icon name="kv-error"></kv-icon>
      <span class="status-label">
        Unknown
      </span>
    </div>
  </mock:shadow-root>
</kv-tag-status>
`;
